code/design_pattern/src/OOP-patterns/strategy/equation.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+

```

Comment:
I would prefer normal header guards as well, since `#pragma once` is _still_ nonstandard, which could possibly cause a problem. Use both kinds of guards.